modules/adblocker/sources/background.es

```diff
@@ -243,14 +248,11 @@ export default background({
         return { active: false };
       }
 
-      return new Promise((resolve) => {
-        this.adblocker.manager.engine.handleRuntimeMessage(
-          browser,
-          { action: 'getCosmeticsFilters', ...payload },
-          sender,
-          resolve,
-        ).catch(() => { /* it's ok if this fails */ });
-      });
+      return this.adblocker.manager.engine.onRuntimeMessage(
+        browser,
+        { action: 'getCosmeticsFilters', ...payload },
+        sender
+      );
```

Comment:
Is it correct to assume that this also solves the issue of un-resolved promises? (i.e. https://github.com/ghostery/adblocker/pull/2724)

Reply:
Maybe not solving it, since there are more places as far as I remember. But it is a step in the right direction.

Another reason why I change it is because it was accessing a private function in the adblocker; now it is using a public function.